src/Cli/dotnet/commands/dotnet-new/PostActions/DotnetAddJsonPropertyPostActionProcessor.cs

```diff
@@ -0,0 +1,235 @@
+        {
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+            WriteIndented = true
+        };
+
+        private static readonly JsonDocumentOptions s_deserializerOptions = new JsonDocumentOptions
+        {
+            AllowTrailingCommas = true,
+            CommentHandling = JsonCommentHandling.Skip
+        };
+
+        private const string JsonFileNameArgument = "jsonFileName";
+        private const string ParentPropertyPathArgument = "parentPropertyPath";
+        private const string NewJsonPropertyNameArgument = "newJsonPropertyName";
+        private const string NewJsonPropertyValueArgument = "newJsonPropertyValue";
+
+        protected override bool ProcessInternal(
+            IEngineEnvironmentSettings environment,
+            IPostAction action,
+            ICreationEffects creationEffects,
+            ICreationResult templateCreationResult,
+            string outputBasePath)
+        {
+            if (!action.Args.TryGetValue(JsonFileNameArgument, out string? jsonFileName))
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Error_ArgumentNotConfigured, JsonFileNameArgument));
+                return false;
+            }
+
+            IReadOnlyList<string> jsonFiles = FindFilesAtOrAbovePath(environment.Host.FileSystem, outputBasePath, matchPattern: jsonFileName, maxAllowedAboveDirectories: 1);
+
+            if (jsonFiles.Count == 0)
+            {
+                Reporter.Error.WriteLine(LocalizableStrings.PostAction_ModifyJson_Error_NoJsonFile);
+                return false;
+            }
+
+            if (jsonFiles.Count > 1)
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Error_MultipleJsonFiles, jsonFileName));
+                return false;
+            }
+
+            // If no ParentPropertyPath is specified, the new JSON property must be added to the root of the
+            // document.
+            action.Args.TryGetValue(ParentPropertyPathArgument, out string? parentProperty);
+            action.Args.TryGetValue(NewJsonPropertyNameArgument, out string? newJsonPropertyName);
+            action.Args.TryGetValue(NewJsonPropertyValueArgument, out string? newJsonPropertyValue);
+
+            if (newJsonPropertyName == null)
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Error_ArgumentNotConfigured, NewJsonPropertyNameArgument));
+                return false;
+            }
+
+            if (newJsonPropertyValue == null)
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Error_ArgumentNotConfigured, NewJsonPropertyValueArgument));
+                return false;
+            }
+
+            var sourceNameReplaceValue = DetermineSourceNameReplaceValue(creationEffects);
```

Comment:
The logic of determining the `sourceName` is not fully correct.
I would recommend to consider replacing all file renames for this properties, or better create an additional configuration which file renames to consider.
This configuration should contain the list of source filenames, check the final names for those and apply renames to `newJsonPropertyName` and `newJsonPropertyValue`

Reply:
I'm afraid I don't follow here.

What I was actually trying to do here, was to make sure that, if the `newJsonPropertyName` or `newJsonPropertyValue` is set to the value of the `sourceName` property in the template.json, the `newJsonPropertyValue` or `newJsonPropertyName` is set to the value that is overriden by the -n or --name parameter that was used in the dotnet new command.